doc/faq.rst

```diff
@@ -350,6 +350,9 @@ improvements, if any, with benchmarks and/or plots. It is expected that the
 proposed algorithm should outperform the methods that are already implemented
 in scikit-learn at least in some areas.
 
+Please do not propose algorithms you (your best friend, colleague or boss)
```

Comment:
I wonder if it would be useful to add your second point, that they can "contribute to" any existing discussion and implementation already underway. But maybe that is obvious...?